apiserver/cloud/instance_information.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package cloud
+
+import (
+	"github.com/juju/errors"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/constraints"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/state/stateenvirons"
+)
+
+// EnvironConfigGetter implements environs.EnvironConfigGetter
+// in terms of a *state.State.
+type cloudEnvironConfigGetter struct {
+	Backend
+	region string
+}
+
+// CloudSpec implements environs.EnvironConfigGetter.
+func (g cloudEnvironConfigGetter) CloudSpec(tag names.ModelTag) (environs.CloudSpec, error) {
+	model, err := g.GetModel(tag)
+	if err != nil {
+		return environs.CloudSpec{}, errors.Trace(err)
+	}
+	cloudName := model.Cloud()
+	regionName := g.region
+	credentialTag, _ := model.CloudCredential()
+	return stateenvirons.CloudSpec(g.Backend, cloudName, regionName, credentialTag)
+}
+
+// InstanceTypes returns instance type information for the cloud and region
+// in which the current model is deployed.
+func (api *CloudAPI) InstanceTypes(cons params.CloudInstanceTypesConstraints) (params.InstanceTypesResults, error) {
+	return instanceTypes(api, environs.GetEnviron, cons)
+}
+
+type environGetFunc func(st environs.EnvironConfigGetter, newEnviron environs.NewEnvironFunc) (environs.Environ, error)
+
+func instanceTypes(api *CloudAPI,
+	environGet environGetFunc,
+	cons params.CloudInstanceTypesConstraints,
+) (params.InstanceTypesResults, error) {
+	c, err := api.backend.ControllerModel()
+	if err != nil {
+		return params.InstanceTypesResults{}, errors.Trace(err)
+	}
+	modelTag := c.ModelTag()
+	m, err := api.backend.GetModel(modelTag)
```

Comment:
you end up with the same model as returned by ControllerModel

just do `m, err := api.backend.ControllerModel()`

then you can get rid of the ModelTag method from Backend